champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/axe/ShieldSmash.java

```diff
@@ -0,0 +1,143 @@
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilVelocity;
+import me.mykindos.betterpvp.core.utilities.events.EntityProperty;
+import org.bukkit.Location;
+import org.bukkit.Particle;
+import org.bukkit.Sound;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.potion.PotionEffectType;
+import org.bukkit.util.Vector;
+
+@Singleton
+@BPvPListener
+public class ShieldSmash extends Skill implements InteractSkill, CooldownSkill, Listener {
+
+    private double baseMultiplier;
+
+    @Inject
+    public ShieldSmash(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Shield Smash";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+
+        return new String[]{
+                "Smash your shield into an enemy",
+                "dealing <val>" + (int) (getKnockbackMultiplier(level) * 100) + "%</val> knockback.",
+                "",
+                "Recharge: <val>" + getCooldown(level)
+        };
+    }
+
+    @Override
+    public Role getClassType() {
+        return Role.KNIGHT;
+    }
+
+    @Override
+    public SkillType getType() {
+        return SkillType.AXE;
+    }
+
+
+    @EventHandler
+    public void onDamage(CustomDamageEvent event) {
+        if (event.isCancelled()) return;
+        if (!(event.getDamagee() instanceof Player player)) return;
+        if (!hasSkill(player)) return;
+
+        if (player.hasPotionEffect(PotionEffectType.DAMAGE_RESISTANCE)) {
+            event.setKnockback(false);
+        }
+    }
+
+    private double getKnockbackMultiplier(int level){
+        return baseMultiplier + ((level - 1) * 0.2);
+    }
+
+    @Override
+    public double getCooldown(int level) {
+        return cooldown - (level - 1d);
+    }
+
+
+    @Override
+    public void activate(Player player, int level) {
+        final Location bashLocation = player.getLocation().add(0, 0.8, 0);
+        bashLocation.add(player.getLocation().getDirection().setY(0).normalize().multiply(1.5));
+
+        // Visual Cues
+        Particle.CLOUD.builder().extra(0.05f).count(6).location(bashLocation).receivers(60, true).spawn();
+        Particle.EXPLOSION_LARGE.builder().extra(0).count(1).location(bashLocation).receivers(60, true).spawn();
+
+        // Skill
+        Vector direction = player.getLocation().getDirection();
+        direction.setY(Math.max(0, direction.getY())); // Prevents downwards knockback
+
+        final double strength = getKnockbackMultiplier(level);
+        final List<KeyValue<LivingEntity, EntityProperty>> bashed = UtilEntity.getNearbyEntities(player, bashLocation, 2.5, EntityProperty.ALL);
```

Comment:
Couldnt you use getNearbyEnemies, which will filter all the other stuff out for you

Reply:
It should work on all entities. People use it on their allies to help them get on higher locations